### ISSUE TITLE
How to use with C# to generate satellite .dll

### ISSUE THREAD

**Issue by Kryptos-FR** [vslavik/gettext-tools-windows#3]
Hi,

I'm currently evaluating tools and framework to localize our application written in C#. Following the [Gettext documentation](https://www.gnu.org/software/gettext/manual/html_node/C_0023.html), I'm trying to use `msgfmt.exe` to generate either `.resources` files or satellite `.dll`.

However the tool complains that i cannot find a C# compiler and ask to install [pnet](https://www.gnu.org/software/dotgnu/pnet-packages.html). The problem is that this .Net compiler seems abandoned and in the best case obsolete (last stable release 10 years ago).

Is there a way to use a more recent compiler with the Gettext tools (Roslyn)? Without proper compiler support it seems pointless to try to use Gettext nowadays.

**edit**
I tried to have the `csc.exe` in the path to be used as C# compiler. Now the issue is that it cannot find the `GNU.Gettext.dll` file. Why is this file not included in the package? And where can I find a compatible one?

> C:\Projects\localization-test\packages\Gettext.Tools.0.19.8.1\tools\bin>msgfmt.exe -v --csharp -l fr-FR -d ..\..\..\..\sources\locale\fr-FR ..\..\..\..\sources\locale\fr-FR\Example.po
csc -target:library '-out:..\..\..\..\sources\locale\fr-FR/fr-FR/Messages.resources.dll' '-lib:C:\Projects\xenko-localization\packages\Gettext.Tools.0.19.8.1\tools/lib' -reference:GNU.Gettext.dll -optimize+ 'C:\Users\KRYPTOS\AppData\Local\Temp\msgXDcP8D/resset.cs'
Microsoft (R) Visual C# Compiler version 2.2.0.61624
Copyright (C) Microsoft Corporation. All rights reserved.
>
> error CS0006: Metadata file 'GNU.Gettext.dll' could not be found
msgfmt.exe: compilation of C# class failed
8 translated messages.

**Comment by vslavik** [comment]
This repository simply hosts — as is rather obvious from its description — win32 binaries. You're crying on the wrong grave. 

> Why is this file not included in the package? 

Your PR addressing this is eagerly awaited.

**Comment by Kryptos-FR** [comment]
Isn't `GNU.Gettext.dll` supposed to be part of the Gettext package? The binaries themselves need this dependency to work.

**Comment by vslavik** [comment]
Depends very much on `configure` switches and detections, as usual.

Again: I would gladly accept PRs improving the build — investigate the issue, work it out, perhaps ask gettext folks. I merely compile the binaries in a standard MinGW environment and have no need for this, so I won't be spending *my* time on this.

**Comment by whut** [comment]
@Kryptos-FR 

Check the [mingw-w64-gettext](https://github.com/Alexpux/MINGW-packages/tree/master/mingw-w64-gettext), I used this to have the basics of "./configure" done already by someone else.

I also had issues with C# compiler (at the time I used GetText 0.19.4, it was almost 2 years ago). Some parts of gettext called csc.exe with linux paths (both during build of GNU.GetText.dll and later when using msgfmt.exe), which where not working for csc.exe from .NET Framework. I fixed this by writing my own csc.exe that just converted the paths and invoked the real csc.exe.
